src/Life/Main/Add.hs

```diff
@@ -45,28 +51,51 @@ lifeAdd lPath = whatIsLife >>= \case
                 filePath <- resolveFile homeDirPath path
                 whenM (doesFileExist filePath) $ do
                     relativeFile <- makeRelative homeDirPath filePath
-                    resolveConfiguration singleFileConfig filename relativeFile
+                    resolveConfiguration files checkEqualFiles copyFileWithDir relativeFile
 
             (Dir path)  -> do
                 dirPath <- resolveDir homeDirPath path
                 whenM (doesDirExist dirPath) $ do
                     relativeDir <- makeRelative homeDirPath dirPath
-                    resolveConfiguration singleDirConfig dirname relativeDir
+                    resolveConfiguration directories checkEqualDirs copyDirRecur relativeDir
 
             -- We didn't find the file
         errorMessage "The file/directory doesn't exist" >> exitFailure
 
-
-resolveConfiguration :: (Path Rel t -> LifeConfiguration)
-                     -> (Path Rel t -> Path Rel t)
+resolveConfiguration :: Lens' LifeConfiguration (Set (Path Rel t))
+                     -> (Path Rel t -> IO Bool)
+                     -> (Path Abs t -> Path Abs t -> IO ())
                      -> Path Rel t
                      -> IO ()
-resolveConfiguration configBuilder pathName path = do
+resolveConfiguration confLens checkContent copyFun path = do
     configuration <- parseGlobalLife
-    let newConfiguration = configuration <> configBuilder path
-    writeGlobalLife newConfiguration
+    let newConfiguration = configuration & confLens %~ Set.insert path
+
+    isSameAsInRepo <- checkContent path
+    if isSameAsInRepo then do
+        let pathText = toText $ toFilePath path
+        infoMessage $ "Path " <> pathText <> " is already latest version in repository"
+    else do
+        writeGlobalLife newConfiguration
+        addToRepo copyFun path
 
-    let pathTextName = toText $ toFilePath $ pathName path
-    let commitMsg    = "Add: " <> pathTextName
-    updateDotfilesRepo commitMsg newConfiguration
     exitSuccess
+
+checkEqualFiles :: Path Rel File -> IO Bool
+checkEqualFiles path = do
+    homeFilePath <- relativeToHome path
+    repoFilePath <- relativeToHome (repoName </> path)
+
+    originContent <- LBS.readFile $ toFilePath homeFilePath
+    repoContent <- LBS.readFile $ toFilePath repoFilePath
+
+    pure $ originContent == repoContent
+
+checkEqualDirs :: Path Rel Dir -> IO Bool
+checkEqualDirs _ = do
+    warningMessage "TODO: check directories to be equal"
```

Comment:
there is no such function in the lib?

Reply:
@vrom911 Unfortunately, no 😞 I can use this function to take list of all files and directories:

* https://hackage.haskell.org/package/path-io-1.3.3/docs/Path-IO.html#v:listDirRecur

But also I need to check for empty directories. Because `git` ignores those directories. And I need to check file content pairwise. So writing this function is slightly painful...

Reply:
Yes, I see now..